src/three-components/PMREMGenerator.ts

```diff
@@ -234,32 +265,114 @@ void main() {
   }
 }
 
-class PackingShader extends RawShaderMaterial {
+// This is a hack because IE claims that a shader which unrolls a loop to access
+// 96 texture lookups has "complexity which exceeds allowed limits", however 48
+// seems to be fine, so we subsample. This could be improved if someone cares a
+// lot about IE.
```

Comment:
I wonder if there are low-end devices that might also have issues with this.

Separately, it might be cleaner to read if this was written as a [preprocessor directive](https://threejs.org/docs/#api/en/materials/ShaderMaterial.defines). Although as I write that, I'm not 100% sure if that will prevent IE from barfing when linking the shader.

Reply:
That's a good idea; I'll give that a try.

Reply:
Was it too cumbersome to use Three.js' `defines` property for `ShaderMaterial` to do this?

Reply:
Yeah, those are really for defining constants, not just a switch. I noticed this is the way three tends to do it internally.